### PR TITLE
Remove advanced network policy guide

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 1]
```diff
@@ -47,184 +42,158 @@ container to network AB.
...
+associated with that network.  One example applies policy directly through the
+profile, the other applies policy through label selection.  Both approaches
+may be used together - in which case the selector based policy is applied first.
+
+### Global policy applied through label selection
 
-#### Create a Docker network
+Suppose we want to provide network policy for a set of database containers and a
+set of app containers, restricting the connectivity to allow only
+TCP connections from the app containers to the database containers connecting to
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: I think this should go right before the "Global policy applied" title instead of after it. It'd be more clear to introduce the scenario, then show how each policy approach enforces it

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 2]
```diff
@@ -47,184 +42,158 @@ container to network AB.
...
+Suppose we want to provide network policy for a set of database containers and a
+set of app containers, restricting the connectivity to allow only
+TCP connections from the app containers to the database containers connecting to 
+port 3306.
 
-To create a Docker network using Calico, run the `docker network create`
-command specifying `calico` as the network driver and `calico-ipam` as the IPAM driver.
+One way to do this would be to define two Docker networks, one for apps and the
+other for databases.  By modifying the profiles associated with each network, we
+can apply labels to endpoints in each network and then create some policy rules
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: This point needs to be emphasized more. The whole basis of this  approach is to leverage profiles just to apply labels.

I'm thinking it may make more sense to introduce Profile based policy first, point out its limitations, and then move on to how to do label-based policy instead

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 3]
```diff
@@ -2,37 +2,31 @@
...
-> Note that if you want to access the feature rich Calico policy, you must use
-> both the Calico Network _and_ Calico IPAM drivers together.  Using the Calico
-> IPAM driver ensures _all_ traffic from the container is routed via the host
-> vRouter and is subject to Calico policy.
-
-> The profile that is created by the Calico network driver is given the same 
-> name as the Docker network name.
+A Docker network created using Calico has an identically name Calico profile 
+associated with it which the Calico plugin applies to each endpoint in the network.
+If the profile does not exist the Calico plugin will automatically create it with
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: Leave this out, I think it's an answer to a question most people won't be asking when they read this

Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: Actually, just reword it to say, "By default, Calico creates a profile with a matching name as the docker network with policy rules which...". Less emphasis on "if it doesn't create, create it", and more emphasis on "default rules for a profile"

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 4]
```diff
@@ -2,37 +2,31 @@
...
-
-When creating a Docker network using the Calico network driver, the Calico 
-driver creates a profile object for that network.  The default policy applied 
-by Calico when a new network is created allows communication between all 
-containers connected to that network, and no communication from other networks.
+With Calico as a Docker network plugin, network policy may be applied in two ways:
 
-Using the standard `calicoctl create` to create a `profile` resource, it is possible to 
-manage the feature-rich policy associated with a network.
+-   Policy is defined at the network scope and applied directly to each endpoint
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: I don't think these definitions nail it on the head, I think some more background is required to clear confusion.

Use of `network scope` seems too ambiguous. Its not clear how abstract you are being, whether we're talking about profiles, docker networks, and networking, 

My suggestion for clearing it up would be to do more introduction on calico's network model and libnetwork's network model first, then move onto label-based policy. Here's my crack at it:

---------------

With Calico as a Docker Network plugin,
Calico will create an identically named Profile to represent each  
Docker network. This profile is applied to each container launched
in the network. Calico limits each container to one network / profile. The default rules for this profile allow full communication between endpoints within the same network / profile, and prohibit inboudn traffic from any other source. 

>Note: You may assign custom default rules to a profile
simply by manually creating the profile in advance.

Once your containers are assigned to these profiles, there
are two ways which network policy can be applied.

- a.) Directly modify policy for the profile, which is applied directly to each endpoint in the network. This approach is simple, but 
not very flexible, as each profile must describe the full
set of rules that apply to containers in it.

- b.) Assign labels to the profile, which are in turn assigned
to applications belonging to it. Then, a global set of policies can be defined that uses selectors to determine which subset of the policy is applied to those endpoints based on the labels.

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 5]
```diff
@@ -47,184 +41,163 @@ container to network AB.
...
-        selector: type=='application'
-        tag: alphatag
-      icmp:
-        code: 6
-        type: 10
-      notICMP:
-        code: 255
-        type: 19
-      notProtocol: udp
+    - action: allow
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: For completeness, can we deny outbound from the database?

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 6]
```diff
@@ -47,184 +41,163 @@ container to network AB.
...
+        -  3306
+- apiVersion: v1
+  kind: policy
+  metadata:
+    name: apps
+  spec:
+    order: 0
+    selector: role == 'app'
+    egress:
+    - action: allow
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: For completeness, can we only allow traffic destined to the designated port of the database?

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 7]
```diff
@@ -47,184 +41,163 @@ container to network AB.
...
+EOF
 ```
 
-As you can see, the default rules allow all outbound traffic and accept inbound
-traffic only from containers attached the "databases" network.
+With the policy created, containers created in these two networks will have their
```
Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: The driving point on why this is a better method is that we can reuse this policy in a different network/profile without having to redefine it all over again, like you would with profiles.

A tiny tiny example here explaining that would make it clear. 

-------------------

Let's say I want to deploy a second frontend, and this `new-frontend` has the exact same functionality as the original, but also needs to access an offsite storage database. I can create a new docker network for this frontend, and again assign the `role=app` label to the profile, reusing the already defined policy. I can then assign a new `version=new` label, and a policy that just allows connections to my offsite database.

Comment (djosborne) [inline master/getting-started/docker/tutorials/advanced-policy.md]: This is the main reason I think we should introduce profile policy first. We can end that section in a blurb like this instead, and then segway into how label-based policy fixes it.

**Comment by djosborne** [comment]
Back to you @robbrockbank

File: master/getting-started/docker/tutorials/advanced-policy.md (hunk 5 above)
Comment (robbrockbank) [inline master/getting-started/docker/tutorials/advanced-policy.md]: yes we can - since contract will take care of the egress traffic

File: v2.0/getting-started/docker/tutorials/advanced-policy.md [hunk 8]
```diff
@@ -2,229 +2,286 @@
...
+For your database containers that also need to be on the backup network, use the 
+"database-backup" network.  Since containers in this network will have the two
+labels assigned to it, they will pick up policy that selects both labels - in
+other words they will have the locked down database access plus access to the
+backup network.
+
+Obviously, the example of allowing full access between everything on the "backup"
+network is probably a little too permissive, so you can lock down the access within
+the backup network by modifying the global policy selected by the `backup = true`
+label.
```
Comment (robbrockbank) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: I think multiple label is worth discussing especially WRT to overlapping networks.... but I wonder if I should beef this up with more sample YAML, or leave as is.  Was worried about this getting too long.   I suppose I could split into two docs profile based and selector based....   WDYT?

File: v2.0/getting-started/docker/tutorials/advanced-policy.md [hunk 9]
```diff
@@ -2,229 +2,292 @@
...
-are encapsulated in a Calico "profile".
+With Calico as a Docker network plugin, Calico will create an identically named 
+[profile]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/profile)
+to represent each Docker network.  This profile is applied to each container 
+in that network and the profile is used by Calico to configure access policy
+for that container.  By default, the profile contains rules that allow full
+egress traffic but allow ingress traffic only from containers within the same
+network and no other source.
+
+> Custom policy for a network can be configured by creating in advance, or editing,
```
Comment (djosborne) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: I'd just move this into the previous paragraph as:

Users can manually choose different default rules by creating the profile in advance using Calico.

File: v2.0/getting-started/docker/tutorials/advanced-policy.md [hunk 10]
```diff
@@ -2,229 +2,292 @@
...
 
-
 ## Background
 
-With Calico, a Docker network represents a logical set of rules that define the 
-allowed traffic in and out of containers assigned to that network.  The rules
-are encapsulated in a Calico "profile".
+With Calico as a Docker network plugin, Calico will create an identically named 
+[profile]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/profile)
+to represent each Docker network.  This profile is applied to each container
```
Comment (djosborne) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: Something to think about: This section has turned into Calico-Docker 101. I wonder if we should flesh this out more with sample output and then move  it to  `getting-started/docker/index`. Then make this doc strongly recommend you read that other part first?

Comment (djosborne) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: We can save a move like that for after this PR

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 11]
```diff
@@ -2,229 +2,292 @@
...
-> vRouter and is subject to Calico policy.
+In both cases we create a Calico-Docker network and use the `calicoctl` tool to
+achieve the required isolation.
 
-> The profile that is created by the Calico network driver is given the same 
-> name as the Docker network name.
+The example we consider is this:  We want to provide isolation between a set of 
+database containers and a set of app containers, allowing only inbound
+TCP connections from the app containers to the database containers on
+port 3306.  We also want to allow all traffic between database containers.
```
Comment (robbrockbank) [inline master/getting-started/docker/tutorials/advanced-policy.md]: IMO there is a balance between a fully engineered solution which is gonna get quite verbose or something simple.   Certainly there should be some interconnectivity between the DB containers usually (if the DB is a cluster for example)

File: master/getting-started/docker/tutorials/advanced-policy.md [hunk 12]
```diff
@@ -2,229 +2,292 @@
...
 
-## Multiple networks
+### Policy applied directly by the profile
```
Comment (robbrockbank) [inline master/getting-started/docker/tutorials/advanced-policy.md]: Sounds good

File: v2.0/getting-started/docker/tutorials/advanced-policy.md (hunk 9 above)
Comment (robbrockbank) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: yeah i was mulling over that myself.  you've tipped my balance

File: v2.0/getting-started/docker/tutorials/advanced-policy.md [hunk 13]
```diff
@@ -2,229 +2,292 @@
...
-    - tag2s
+        selector: role == 'database'
+    egress:
+    - action: allow
+      destination:
+        selector: role == 'database'
+- apiVersion: v1
+  kind: profile
+  metadata:
+    name: apps
```
Comment (robbrockbank) [inline v2.0/getting-started/docker/tutorials/advanced-policy.md]: yes yes yes!   good idea ... i don't know why i always choose app since i always end up changing it!

File: master/getting-started/docker/tutorials/advanced-policy.md (hunk 6 above)
Comment (robbrockbank) [inline master/getting-started/docker/tutorials/advanced-policy.md]: yeah i was trying to keep it simple, but why not!  i'll add.
